sync/threadgroup_test.go

```diff
@@ -173,6 +166,40 @@ func TestThreadGroupRace(t *testing.T) {
 	}
 }
 
+func TestThreadGroupClosedOnStop(t *testing.T) {
+	var tg ThreadGroup
+	var closed bool
+	tg.OnStop(func() { closed = true })
+	if closed {
+		t.Fatal("close function should not have been called yet")
+	}
+	if err := tg.Stop(); err != nil {
+		t.Fatal(err)
+	}
+	if !closed {
+		t.Fatal("close function should have been called")
+	}
+
+	// Stop has already been called, so the close function should be called
+	// immediately
+	closed = false
+	tg.OnStop(func() { closed = true })
+	if !closed {
+		t.Fatal("close function should have been called immediately")
+	}
+
+	// this close function (which calls a goroutine) should also be called
+	// immediately since Stop has been called already
+	closedChan := make(chan struct{})
+	tg.OnStop(func() { go func() { closedChan <- struct{}{} }() })
+	select {
+	case <-closedChan:
+		break
+	case <-time.After(10 * time.Second):
```

Comment:
there should be no `time.Sleep` in a short test. The short tests are meant to be well under a second per package. Better to just make this a long test.